### PR TITLE
Summary functions

### PR DESCRIPTION
**Pull request by GearsAD**
- New structures `AbstractDFGVariable`s and `AbstractDFGFactor`s that contain the first-class citizens of a variable and factor respectively (label, tags, timestamp, estimates)
- getSummary(): Returns a summary structure containing the list of `AbstractDFGVariable`s and `AbstractDFGFactor`s
- getSummaryGraph(): Returns a LightDFG graph containing the list of `AbstractDFGVariable`s and `AbstractDFGFactor`s

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -29,6 +29,7 @@ julia = "0.7, 1"
 
 [extras]
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+GraphPlot = "a2cc645c-3eea-5389-862e-a155d0052231"
```
Comment (Affie) [inline Project.toml]: Should GraphPlot be in deps or extras? 
Same with Neo4j. 
If we don't want it to be a dependency and use it with @require it makes sense not to add it in deps?

File: src/LightDFG/FactorGraphs/FactorGraphs.jl [hunk 2]
```diff
@@ -31,19 +31,22 @@ export
     filter_vertices,
     reverse
 
-import DistributedFactorGraphs: DFGNode
-const AbstractNodeType = DFGNode
+# import DistributedFactorGraphs: DFGNode
+# const AbstractNodeType = DFGNode
+import DistributedFactorGraphs: AbstractDFGVariable, AbstractDFGFactor
+const AbstractVariableType = AbstractDFGVariable
+const AbstractFactorType = AbstractDFGFactor
```
Comment (Affie) [inline src/LightDFG/FactorGraphs/FactorGraphs.jl]: This is the way I wanted to do it from the start. So I'm more than happy with the new Abstract types

File: src/LightDFG/services/LightDFG.jl [hunk 3]
```diff
@@ -56,7 +56,7 @@ end
     $(SIGNATURES)
 Add a DFGFactor to a DFG.
 """
-function addFactor!(dfg::LightDFG, variables::Vector{DFGVariable}, factor::DFGFactor)::Bool
+function addFactor!(dfg::LightDFG{<:AbstractParams, V, F}, variables::Vector{V}, factor::F)::Bool where {V <: AbstractDFGVariable, F <: AbstractDFGFactor}
```
Comment (Affie) [inline src/LightDFG/services/LightDFG.jl]: To force one type in a graph

File: src/entities/DFGVariable.jl [hunk 4]
```diff
@@ -130,3 +130,21 @@ internalId(v::DFGVariable) = v._internalId
...
+    $(SIGNATURES)
+Structure for first-class citizens of a DFGVariable.
+"""
+mutable struct DFGVariableSummary <: AbstractDFGVariable
+    label::Symbol
+    timestamp::DateTime
+    tags::Vector{Symbol}
+    estimateDict::Dict{Symbol, Dict{Symbol, VariableEstimate}}
+    _internalId::Int64
+end
```
Comment (Affie) [inline src/entities/DFGVariable.jl]: Big Data

Comment (Affie) [inline src/entities/DFGVariable.jl]: Small Data

File: test/interfaceTests.jl [hunk 5]
```diff
@@ -259,15 +260,42 @@ end
...
+        end
     end
+    for f in lsf(summaryGraph)
+        for field in factorFields
+            @test getfield(getFactor(dfg, f), field) == getfield(getFactor(summaryGraph, f), field)
+        end
+    end
+end
+
+@testset "Producing Dot Files" begin
```
Comment (Affie) [inline test/interfaceTests.jl]: @GearsAD Please have a look if you are ok with this simpler graph for testing toDot

File: test/sandbox.jl [hunk 6]
```diff
@@ -0,0 +1,24 @@
+using Revise
```
Comment (Affie) [inline test/sandbox.jl]: Did you mean to commit this file?

File: src/services/AbstractDFG.jl [hunk 7]
```diff
@@ -444,13 +470,14 @@ Returns state of vertex data `.initialized` flag.
 
 Notes:
 - used by both factor graph variable and Bayes tree clique logic.
-TODO: Refactor
 """
 function isInitialized(var::DFGVariable; key::Symbol=:default)::Bool
-  return var.solverDataDict[key].initialized
+  solverData(var, key) && return solverData(var, key).initialized
```
Comment (Affie) [inline src/services/AbstractDFG.jl]: @GearsAD This causes an error

Comment (Affie) [reply src/services/AbstractDFG.jl]: data = solverData(var, key)
  data == nothing && return false
  return solverData(var, key).initialized

File: src/LightDFG/FactorGraphs/FactorGraphs.jl (hunk 2 above)
Comment (GearsAD) [reply src/LightDFG/FactorGraphs/FactorGraphs.jl]: Cool!

File: src/entities/DFGVariable.jl (hunk 4 above)
Comment (GearsAD) [reply src/entities/DFGVariable.jl]: Yep, but in next PR tomorrow, get this merged tomorrow morning so we can continue, then data as another.

File: Project.toml (hunk 1 above)
Comment (GearsAD) [reply Project.toml]: Yep!

File: src/services/AbstractDFG.jl (hunk 7 above)
Comment (GearsAD) [reply src/services/AbstractDFG.jl]: Yeah I'm doing Python code accidentally again

File: test/interfaceTests.jl (hunk 5 above)
Comment (GearsAD) [reply test/interfaceTests.jl]: Cool, LGTM!

File: test/sandbox.jl (hunk 6 above)
Comment (GearsAD) [reply test/sandbox.jl]: Inadvertent, but doesn't do any harm.

File: src/LightDFG/services/LightDFG.jl (hunk 3 above)
Comment (GearsAD) [reply src/LightDFG/services/LightDFG.jl]: Looks good, nice

**Pull request merged**